cypress/tests/resource-provider-card.spec.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  PVC_NAME,
+  DATA_SOURCE_NAME_NSFS,
+  DATA_SOURCE_NAME_AWS,
+} from '../constants/tests';
+import { dataSourceAWS, dataSourceNSFS } from '../mocks/data-source';
+import { MCGMSCommon } from '../views/mcg-ms-common';
+
+describe('admin dashboard - resource provider card', () => {
+  before(() => {
+    cy.login();
+  });
+
+  after(() => {
+    cy.logout();
+  });
+
+  beforeEach(() => {
+    MCGMSCommon.visitMcgMsDashboard();
+  });
+
+  it('should check if resource provider card is present', () => {
+    // initially when there's no data source created -> no available providers
+    cy.get(`[data-test]="nb-resource-providers-card"`)
+      .should('exist')
+      .contains('Not available');
+  });
+
+  it('should check count of providers when data source is not in a ready state', () => {
+    cy.exec(
+      `echo '${JSON.stringify(
+        dataSourceAWS(DATA_SOURCE_NAME_AWS)
+      )}' | oc create -f -`
+    ).then(() => {
+      cy.byTestID('nb-resource-providers-card').contains('Not available');
```

Comment:
the wrong item is being queried. please add a data-test [here](https://github.com/avanthakkar/mcg-ms-console/blob/8499327164665c01e81b7a5896d438f471eb2e1b/packages/components/dashboard/resource-providers-card/resource-providers-card.tsx#L62) and query and check whether it's not available or not